clair/api.go

```diff
@@ -26,22 +28,40 @@ type apiV3 struct {
 	client clairpb.AncestryServiceClient
 }
 
-func newAPI(url string, version int, timeout time.Duration) (API, error) {
+func newAPI(clair_url string, version int, timeout time.Duration) (API, error) {
 	if version < 3 {
-		return newAPIV1(url, timeout), nil
+		return newAPIV1(clair_url, timeout), nil
 	}
-	return newAPIV3(url)
+	return newAPIV3(clair_url)
 }
 
-func newAPIV1(url string, timeout time.Duration) *apiV1 {
-	if !strings.HasPrefix(url, "http://") && !strings.HasPrefix(url, "https://") {
-		url = fmt.Sprintf("http://%s", url)
+func newAPIV1(clair_url string, timeout time.Duration) *apiV1 {
+	u, err := url.Parse(clair_url)
+	if err != nil {
+		panic(fmt.Sprintf("Cant't parse Clair URL from: %s", clair_url))
 	}
-	if strings.LastIndex(url, ":") < 6 {
-		url = fmt.Sprintf("%s:6060", url)
+
+	host, port, err := net.SplitHostPort(u.Host)
+	if err != nil {
+		panic(fmt.Sprintf("Cant't extract hostname and port from: %s", u.Host))
+	}
+
+	if !(u.Scheme == "http") && !(u.Scheme == "https") {
```

Comment:
would you mind to elaborate why we need this check? also can we have a simpler check ?